solver/src/encoding.rs

```diff
@@ -37,6 +39,51 @@ fn encode_order_flags(order: &UserOrder) -> u8 {
     result
 }
 
+fn encode_uniswap_call(
+    amount_in: U256,
+    amount_out_min: U256,
+    token_in: H160,
+    token_out: H160,
+    payout_to: H160,
+) -> [u8; UNISWAP_INTERACTION_SIZE] {
+    let uniswap = IUniswapV2Router02::at(&dummy::dummy_web3(), H160::zero());
+    let method = uniswap.swap_exact_tokens_for_tokens(
+        amount_in,
+        amount_out_min,
+        vec![token_in, token_out],
+        payout_to,
+        U256::MAX,
+    );
+    let data = method.tx.data.unwrap().0;
+    let mut result = [0u8; UNISWAP_INTERACTION_SIZE];
+    result.copy_from_slice(data.as_slice());
+    result
+}
+
+// To create an ethcontract instance we need to provide a web3 even though we never use it. This
+// module provides a dummy transport and web3.
```

Comment:
This is unfortunate. Maybe we can do some refactoring in `ethcontract` so that this isn't true. I ran into this issue just the other day and was extremely happy to find out that `ethers.js` can create ABI interface types just for encoding and decoding without a provider (Ethers.js lingo for a `Transport`).

I created https://github.com/gnosis/ethcontract-rs/issues/413 to capture this.